test/controllers/IfsControllerSpec.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import models.{FraudRiskRequest, IFRequest, IFRequestPayload, IFRequestPayloadAction}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Result
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import utils.CommonData.ninoMtdIdPairs
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import scala.concurrent.Future
+
+class IfsControllerSpec extends SpecBase{
+
+  private val controller: IfsController = app.injector.instanceOf[IfsController]
+
+
+  private def submitStoreInteraction(calculationId: String): Future[Result] = {
+    val req = IFRequest(
+      serviceRegime = "self-assessment-assist",
+      eventName = "GenerateReport",
+      eventTimestamp = LocalDate.now(),
+      feedbackId = "feedbackId",
+      metadata = List(
+        Map("nino" -> "nino"),
+        Map("taxYear" -> "2023"),
+        Map("calculationId" -> calculationId),
+        Map("customerType" -> "Agent"),
+        Map("agentReferenceNumber" -> "12345"),
+        Map("calculationTimestamp" -> "2023-01-01T18:03:22")
```

Comment:
calculationTimeStamp should be in below format, we can refer code in api just to keep it uniform across
`   {
      "name": "calculationTimestamp",
      "value": "2019-02-15T09:35:15.094Z"
    }`